### DISCUSSION TITLE
ModuleNotFoundError: No module named 'spacy.lemmatizer'

### DISCUSSION THREAD

**Discussion by saipavanmeruga** [explosion/spaCy#8462] [Help: Other Questions] [answered]
## How to reproduce the behaviour

```
      4 import spacy
      5 from spacy.lang.en.stop_words import STOP_WORDS
----> 6 from spacy.lemmatizer import Lemmatizer
      7 from spacy.lookups import Lookups
      8 from cosine_similarity import calculate_cosine_similarity
```
## Your Environment

* Operating System: Windows 10
* Python Version Used:Python 3.9.5
* spaCy Version Used: spacy v3.0.6

**Comment by adrianeboyd** [comment, marked as answer]
It's hard to know for sure without more details, but this code was probably written for spacy v2. You can install the most recent version of v2 with `pip install spacy==2.3.7`. It's usually best to install this in a new virtual environment because pip sometimes doesn't upgrade/downgrade all the dependencies cleanly.

**Reply by Dvvr1227** [reply]
Hi,
I am working on the repo for the project but facing the same error you have answered above...I tried the process you have suggested but couldn't make it up. Down here the error I am mentioning the error I have been facing and the file link in githubfrom spacy.lemmatizer 

**import Lemmatizer
ModuleNotFoundError: No module named 'spacy.lemmatizer'**

repo : https://github.com/lokeshlal/TextToSql/blob/master/main.py

Please help me solve this problem

Thank you

**Reply by adrianeboyd** [reply]
The solution is still the same: install spacy v2 in a new virtual environment.
